products/opensuse/main.pm

```diff
@@ -145,92 +146,107 @@ if (get_var("WITH_UPDATE_REPO")
-        my $de = get_var("DE_PATTERN");
-        loadtest "console/system_prepare";
-        loadtest "console/consoletest_setup";
-        loadtest "console/hostname";
-        loadtest "update/zypper_clear_repos";
-        loadtest "console/install_otherDE_pattern";
-        loadtest "console/consoletest_finish";
-        loadtest "x11/${de}_reconfigure_openqa";
-        loadtest "x11/reboot_icewm";
-        # here comes the actual desktop specific test
-        if ($de =~ /^awesome$/)       { load_awesome_tests(); }
-        if ($de =~ /^enlightenment$/) { load_enlightenment_tests(); }
-        if ($de =~ /^mate$/)          { load_mate_tests(); }
-        if ($de =~ /^lxqt$/)          { load_lxqt_tests(); }
-        load_shutdown_tests;
-        return 1;
-    }
-    return 0;
-}
+  load_module($module)
 
-sub load_awesome_tests {
-    loadtest "x11/awesome_menu";
-    loadtest "x11/awesome_xterm";
-}
+Load one test module from a schedule hash. The inclusion/exclusion of the
+module can be controlled by specifying the module with C<module> and the key
+C<only> or <except> next to the module.
+
+We could have this logic to include/exclude specific modules in the modules itself but this would contradict
+https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/150
+even though we still have C<is_applicable> in basetest.pm in os-autoinst.
+
+=cut
 
-sub load_enlightenment_tests {
-    loadtest "x11/enlightenment_first_start";
-    loadtest "x11/terminology";
+sub load_module {
+    my ($module) = @_;
+    return $module unless ref $module eq 'HASH';
+    my %module = %{$module};
+    return $module{module} if ($module{module} && ($module{only} && eval_yaml($module{only})) && !($module{except} && !eval_yaml($module{except})));
 }
 
-sub load_lxqt_tests {
+=head2 load_schedule
+
+  load_schedule($tests)
+
+Loads test modules or groups of tests by identifier from C<main.yml> in the same directory as the current file.
+
+In the simple case of just scheduling a list of test modules specify them as an array directly:
+
+ my_tests:
+   - module1
+   - module2
+
+or in shorter JSON style:
+
+ my_tests: [module_1, module_2]
+
+More advanced instructions are possible by specifying according keys in the
+section. In this case the test modules must be specified in an array with the
+name C<modules>. The keywords C<only> an C<except> denote a test variable or a
+perl expression that must evaluate to a true respectively false value for the
+section to load. Other sections can be referenced with C<groups>. Test
+variables specified like perl variables are expanded.
+Inclusion/exclusion of single modules within the modules list is also possible
+by specifying the module with C<module> and the key C<only> or <except> next
+to the module.
+
+An advanced example:
+
+ my_tests_advanced:
+   only: is_advanced()
+   except: VAR1
+   modules:
+     - module1
+     - module2
+     - module: module3
+       except: NO_MOD3
+
+=cut
+
+sub load_schedule {
```

Comment:
So if I understand correctly, this load_schedule will behave different depending on whether there's a section called only or if there's an array? Why not simply make it to expect an array by default?